File: support_tools/swigwin-3.0.12/Examples/test-suite/python/struct_value_runme.py
import struct_value

b = struct_value.Bar()

b.a.x = 3
if b.a.x != 3:
    raise RuntimeError

b.b.x = 3
if b.b.x != 3:
    raise RuntimeError


# Test dynamically added attributes - Github pull request #320
b.added = 123

if b.added != 123:
    raise RuntimeError("Wrong attribute value")

if not b.__dict__.has_key("added"):
    raise RuntimeError("Missing added attribute in __dict__")


class PyBar(struct_value.Bar):

    def __init__(self):
        self.extra = "hi"
        struct_value.Bar.__init__(self)

pybar = PyBar()
if not pybar.__dict__.has_key("extra"):
    raise RuntimeError("Missing extra attribute in __dict__")
if pybar.extra != "hi":
    raise RuntimeError("Incorrect attribute value for extra")

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/struct_value_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'struct_value'

b = Struct_value::Bar.new

b.a.x = 3
raise RuntimeError if b.a.x != 3

b.b.x = 3
raise RuntimeError if b.b.x != 3

